rust/osm2lanes/src/transform/tags_to_lanes/separator/semantic.rs

```diff
@@ -0,0 +1,35 @@
+use crate::metric::Metre;
+use crate::road::Style;
+
+/// Semantic speed class
+enum _SpeedClass {
+    Walking,
+    /// ~30kph / ~20mph
+    Living,
+    /// ~50kph / ~30mph
+    Intra,
+    /// ~80kph / ~50mph
+    Inter,
+    Max,
+}
+
+/// Semantic lane separator
+enum _Separator {
+    /// Into grass or dirt
+    SoftEdge,
+    /// Into a building or other hard surface
+    HardEdge,
+    /// Motorway (or other) shoulder
+    Shoulder(_SpeedClass),
+    /// Road paint for same direction
+    Lane { speed: _SpeedClass },
```

Comment:
Is speed class important here? For lane and centre separators, I'd imagine the "can you change lanes here" and "can you over-take (against opposite-flowing traffic)" are the attributes we may want to refine

Reply:
I think in some countries the paint style differs based on speed class, the classic example is multiple lanes on a motorway vs multiple lanes in an inner-city artery?

But agreed, I am missing the overtake part :)

Reply:
Having read #100 more carefully, I understand where this is going and am in favor. :)